doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,418 @@
+.. _cmake_pkg:
+
+Zephyr CMake Package
+####################
+
+Zephyr can be exported as a CMake config package.
+
+A CMake config package allows a developer to integrate Zephyr into their own
+application by using ``find_package(Zephyr)``.
+
+Exporting Zephyr as a CMake package is identical to creating a reference to the
+current Zephyr installation inside the CMake user package registry.
+
+
+.. tabs::
+
+   .. group-tab:: Ubuntu
+
+      In Linux, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: macOS
+
+      In macOS, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: Windows
+
+      In Windows, the CMake user package registry is found in:
+
+      ``HKEY_CURRENT_USER\Software\Kitware\CMake\Packages\Zephyr``
+
+
+The Zephyr package allows CMake to automatically find a Zephyr base.
+One or more Zephyr installations must be exported.
+Exporting multiple Zephyr installations may be useful when developing or testing
+Zephyr freestanding applications, Zephyr workspace application with vendor forks, etc..
+
+
+Zephyr CMake package export (west)
+**********************************
+
+When installing Zephyr using :ref:`west <get_the_code>` then it is recommended
+to export Zephyr using ``west zephyr-export``.
+
+
+Zephyr CMake package export (without west)
+******************************************
+
+Zephyr can be exported using CMake directly.
+To do so, run the following command:
+
+.. code-block:: bash
+
+   cmake -S <PATH-TO-ZEPHYR>/share/zephyr-package/cmake -B <PATH-TO-ZEPHYR>/share/zephyr-package/cmake
+   cmake --build <PATH-TO-ZEPHYR>/share/zephyr-package/cmake --target pristine
+
+This will export the current Zephyr to the CMake user package registry and remove the temporary
+files generated by CMake during export.
+
+.. _zephyr_cmake_package_zephyr_base:
+
+
+Zephyr application structure
+****************************
+
+An application can be placed anywhere on your disk, but to better understand how the Zephyr
+package is used, we will name three specific layouts.
```

Comment:
I'm wondering if there's a difference between a "repository application" and a "workspace application" for the purpose of this document. Can they be merged?

Reply:
There is a difference, especially as the workspace may have multiple Zephyr's to choose from.